nimble/host/mesh/src/prov.c

```diff
@@ -898,24 +898,14 @@ static void send_pub_key(void)
 		goto done;
 	}
 
-	prov_buf_init(buf, PROV_PUB_KEY);
-
-	/* Swap X and Y halves independently to big-endian */
-	sys_memcpy_swap(net_buf_simple_add(buf, 32), key, 32);
-	sys_memcpy_swap(net_buf_simple_add(buf, 32), &key[32], 32);
-
-	memcpy(&link.conf_inputs[81], &buf->om_data[1], 64);
-
-	BT_DBG("Local Public Key: %s", bt_hex(&buf->om_data[1], 64));
-
-	prov_send(buf);
+	/* bt_dh_key_gen() will verify that the remote's public key is valid. */
 
 	/* Copy remote key in little-endian for bt_dh_key_gen().
 	 * X and Y halves are swapped independently.
 	 */
 	net_buf_simple_init(buf, 0);
-	sys_memcpy_swap(buf->om_data, &link.conf_inputs[17], 32);
-	sys_memcpy_swap(&buf->om_data[32], &link.conf_inputs[49], 32);
+	sys_memcpy_swap(net_buf_simple_add(buf, 32), &link.conf_inputs[17], 32);
+	sys_memcpy_swap(net_buf_simple_add(buf, 32), &link.conf_inputs[49], 32);
```

Comment:
Why are you using net_buf_simple_add here? From what I see bt_dh_key_gen() doesn't care about the net_buf_simple internal bookkeeping, rather just gets a pointer to the data. It's only later for the response that the net_buf_simple state needs to be correct, and starting by calling prov_buf_init() should take care of that. In fact, prov_buf_init() does not take care of resetting buf->len so you're ending up with a corrupt (overflowing) buffer as far as I see.

Reply:
I think net_buf_simple_init() does take care of buf->om_len in Mynewt. Just wanted to change that for consistency. I will go back to using buf->om_data directly.

Reply:
But you're not calling net_buf_simple_init() before prov_buf_init() (after having added the above data to the buffer). Anyway, this code is quite different already in Zephyr so I'll let you sort this out.

Reply:
We are calling net_buf_simple_init() in prov_buf_init(), that's why I used it.

Reply:
Ok, makes sense. The Zephyr prov_buf_init() doesn't do that (it doesn't need to). Either way it's unnecessary code cycles to modify the buffer internal state as far as I see, since all you need is a place to temporarily store the 64 bytes.

Reply:
Agreed, I deleted that change.